server/cluster/cluster.go

```diff
@@ -84,12 +111,17 @@ func (cluster *Cluster) Start() error {
 		logger.WithField("remotes", cluster.remotesAsStrings()).Error(errorMessage)
 		return errors.New(errorMessage)
 	}
+
 	logger.Debug("Started Cluster")
+
 	return nil
 }
 
 // Stop the cluster module.
 func (cluster *Cluster) Stop() error {
+	if cluster.synchronizer != nil {
+		close(cluster.synchronizer.stopC)
```

Comment:
you could safely add a line here, to allow multiple Stop() invocations without any crash:
   cluster.synchronizer = nil